backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java

```diff
@@ -487,6 +541,420 @@ public Response getProxyPullRequest(String url, String authKey, String authCode,
+                    log.info("Need Contract for ItemStock Submodel 2.0.0 with " + partner.getBpnl());
+                    if (negotiateForPartnerItemStockSubmodel(mpr, direction)) {
+                        contractMapping = edcContractMappingService.find(partner.getBpnl());
+                        itemStockContractId = contractMapping.getItemStockContractId();
+                        assetId = contractMapping.getItemStockEdcProtocolUrl();
+                        partnerItemStockEdcUrl = contractMapping.getItemStockEdcProtocolUrl();
+                        href = contractMapping.getMaterialToHrefMapping().get(materialNumber);
+                    } else {
+                        log.error("Failed to contract for ItemStock Submodel 2.0.0 with " + partner.getBpnl());
+                        return null;
+                    }
+                }
+                try {
+                    if (!partnerItemStockEdcUrl.equals(partner.getEdcUrl())) {
+                        log.warn("Diverging EDC URL's for partner " + partner.getBpnl());
+                        log.warn("Edc URL from AAS " + partnerItemStockEdcUrl);
+                        log.warn("General Edc Url of partner " + partner.getEdcUrl());
+                    }
+                    // Request EdrToken
+                    var transferResp = initiateProxyPullTransfer(partner, itemStockContractId, assetId, partnerItemStockEdcUrl);
+                    String transferId = transferResp.get("@id").asText();
+                    for (int i = 0; i < 100; i++) {
+                        Thread.sleep(100);
+                        transferResp = getTransferState(transferId);
+                        if ("STARTED".equals(transferResp.get("edc:state").asText())) {
+                            break;
+                        }
+                    }
+                    EDR_Dto edrDto = null;
+                    // Await arrival of edr
+                    for (int i = 0; i < 100; i++) {
+                        Thread.sleep(100);
+                        edrDto = edrService.findByTransferId(transferId);
+                        if (edrDto != null) {
+                            log.info("Received EDR data for " + assetId + " with " + partner.getEdcUrl());
+                            break;
+                        }
+                    }
+                    if (edrDto == null) {
+                        log.error("Failed to obtain EDR data for " + assetId + " with " + partner.getEdcUrl());
+                        return null;
+                    }
+                    var edrURL = edrDto.endpoint();
+                    if (!edrURL.endsWith("/")){
+                        edrURL += "/";
+                    }
+                    edrURL += materialNumber + "/" + direction;
+                    if (href.endsWith("/")) {
+                        edrURL+= "/";
+                    }
+                    if (!edrURL.equals(href)) {
```

Comment:
Why are we concatenating the edrUrl with materialNumber and direction? The concatenated edrUrl will be the same for the same version of the PURIS FOSS application but not (for sure) for other implementations as one can use the href for different implementations as long as one can just append /$value. 

Wouldn't it be sufficient to check for !href.startsWith(edrUrl)? In case the public api of the data plane changed, we can't derive the right url for every partner and can log an error and stop.

Reply:
You're right. I simplied this part a bit.